commons/pom.xml

```diff
@@ -17,6 +17,7 @@
       <dependency>
          <groupId>com.github.ben-manes.caffeine</groupId>
          <artifactId>caffeine</artifactId>
+         <optional>true</optional>
```

Comment:
Although I am not sure if the CollectionFactory class itself would work since it has an import for Caffeine.